### ISSUE TITLE
RegexpSingleline Check Fails to Escape '\n' in Message, Causing Broken JSON Output Format

### ISSUE THREAD

**Issue by Abdelrhmansersawy** [checkstyle/checkstyle#16743]
While solving the following PR: [PR #16698](https://github.com/checkstyle/checkstyle/pull/16698).
I noticed that when using the `RegexpSingleline` check with a message containing `\n`, the Checkstyle output in JSON format is broken. Specifically, the newline character `\n` is not properly escaped, causing the JSON parser to fail. The character `\n` is escaped as `\\n`, which is incorrect and breaks the JSON format.

---

### **Expected Behavior:**

The message in the JSON output should properly display the newline character as `\n`:

```json
"text": "Copyright comment found. \n try again"
```

---

### **Actual Behavior:**

Instead, the output JSON incorrectly displays the `\n` character as `\\n`, which causes issues with JSON parsing:

```json
"text": "Copyright comment found. \\n try again"
```

---

### **Steps to Reproduce:**

1. Create the following Checkstyle configuration in `config.xml`:

   ```xml
   <module name="Checker">
     <module name="TreeWalker">
       <module name="RegexpSingleline">
         <property name="format" value="// This code is copyrighted\."/>
         <property name="message" value="Copyright comment found. \n try again"/>
       </module>
     </module>
   </module>
   ```

2. Use the following Java class (`InputSarifLoggerErrorColumn0.java`) from the PR:

   [Java Class Example](https://github.com/checkstyle/checkstyle/blob/9c9b9916f76786eaebdf0f287c8c9fa467e220dc/src/test/resources/com/puppycrawl/tools/checkstyle/sariflogger/InputSarifLoggerErrorColumn0.java)

   ```java
   package com.puppycrawl.tools.checkstyle.sariflogger;
   // This code is copyrighted.
   public class InputSarifLoggerErrorColumn0 {
   }
   ```

3. When writing this Java class into the PR, I found the parsed SARIF output to be incorrect:

   [SARIF Output Example](https://github.com/checkstyle/checkstyle/blob/9c9b9916f76786eaebdf0f287c8c9fa467e220dc/src/test/resources/com/puppycrawl/tools/checkstyle/sariflogger/ExpectedSarifLoggerSingleErrorColumn0.sarif)

   The generated SARIF output is:

   ```json
   {
     "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
     "version": "2.1.0",
     "runs": [
       {
         "tool": {
           "driver": {
             "downloadUri": "https://github.com/checkstyle/checkstyle/releases/",
             "fullName": "Checkstyle",
             "informationUri": "https://checkstyle.org/",
             "language": "en",
             "name": "Checkstyle",
             "organization": "Checkstyle",
             "rules": [],
             "semanticVersion": "null",
             "version": "null"
           }
         },
         "results": [
           {
             "level": "error",
             "locations": [
               {
                 "physicalLocation": {
                   "artifactLocation": {
                     "uri": "file:src/test/resources/com/puppycrawl/tools/checkstyle/sariflogger/InputSarifLoggerErrorColumn0.java"
                   },
                   "region": {
                     "startLine": 13
                   }
                 }
               }
             ],
             "message": {
               "text": "Copyright comment found. \\n try again"
             },
             "ruleId": "regexp.exceeded"
           }
         ]
       }
     ]
   }
   ```

   Notice the incorrect escape of `\n` as `\\n` in the `message` field of the SARIF output.

**Comment by romani** [comment]
https://stackoverflow.com/a/42073/1015848 people thinks that double slash before `n` is correct way .

**Comment by romani** [comment]
Referenced PR is merged, issue is not valid
